### PR TITLE
Refactor old accounting classes into DTOs. 

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Purpose of this PR**
This PR deprecate the old accounting `CostEstimate` class and replace it with a DTO akin to the` Offer `DTO.
Additionally it moves the DTOs `ProductItem `and `ProductCategory` into the dto/business repository.
Lastly this PR updates the DTO descriptions and the Tests to the current DTOs

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/accounting/Product.groovy [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package life.qbic.datamodel.accounting
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/accounting/Product.groovy:1]: isnt Product etc not also a DTO?

Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/accounting/Product.groovy:1]: Ah its the old and deprecated Product class, is it?

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        Customer customer
+        ProjectManager projectManager
+        String projectTitle
+        String projectDescription
+        List<ProductItem> items
+        double totalPrice
+        CostEstimateId identifier
+        Affiliation selectedCustomerAffiliation
+
+        Builder(Date modificationDate, Date expirationDate, Customer customer, ProjectManager projectManager, String projectDescription, String projectTitle, List<ProductItem> items, double totalPrice, CostEstimateId identifier, Affiliation selectedCustomerAffiliation) {
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy:73]: Make sure this is exchanged with the constructor you drafted in your other PR :)

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
What is about this Product class, is it the old one? If yes LGTM otherwise I would ask you to also move it to the DTOs

File: src/main/groovy/life/qbic/datamodel/accounting/Product.groovy (hunk 1 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/accounting/Product.groovy:1]: Yes :)

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy (hunk 2 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy:73]: Will do! Thanks for reminding me!

Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy]: Done!

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy [hunk 3]
```diff
@@ -1,37 +1,33 @@
...
+ * A cost estimate for a project
  *
  * During project planning a cost estimate is often required to list expected costs.
  * This estimate is not intended to be legally binding. It provides an overview over the project
  * scope and the costs associated with it.
  *
  * @since: 1.12.0
- *
  */
-@EqualsAndHashCode
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy]: why deleting this?

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy [hunk 4]
```diff
@@ -97,11 +98,6 @@ class CostEstimate {
             return this
         }
 
-        Builder selectedCustomerAffiliation(Affiliation selectedCustomerAffiliation) {
-            this.selectedCustomerAffiliation = selectedCustomerAffiliation
-            return this
-        }
-
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy:104]: wait can we not add the selected customer affiliation to the object now?

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy (hunk 3 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy]: Oversight during merge conflict. Nice Catch!

File: src/test/groovy/life/qbic/datamodel/dtos/business/CostEstimateSpec.groovy [hunk 5]
```diff
@@ -46,9 +46,6 @@ class CostEstimateSpec extends Specification {
 
     def "Missing optional Field definitions shall haven null values in a CostEstimate object"() {
 
-        given:
-        CostEstimateId costEstimateId = new CostEstimateId("ab", "cd", 1)
-
         when:
         Offer testOffer =
                 new Offer.Builder(customer, projectManager, "Archer", "Cartoon Series", [], selectedAffiliation).build()
```
Comment (jenniferboedker) [inline src/test/groovy/life/qbic/datamodel/dtos/business/CostEstimateSpec.groovy:51]: why do we have an offer here and not a cost estimate?

File: src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy (hunk 4 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/dtos/business/CostEstimate.groovy:104]: No it was a relict of the DTO in which only the `modificationDate` was required. 
I adapted the DTO so the `customerAffiliation` has to be provided during the DTO generation and is not optional.

File: src/test/groovy/life/qbic/datamodel/dtos/business/CostEstimateSpec.groovy (hunk 5 above)
Comment (Steffengreiner) [reply src/test/groovy/life/qbic/datamodel/dtos/business/CostEstimateSpec.groovy]: Good Catch, i adapted the test and oversaw that line 👍

**Review by jenniferboedker — APPROVED** [review]
Looks good now! Thanks for being patient with my comments and NITS

File: src/main/groovy/life/qbic/datamodel/accounting/Product.groovy [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
+
 /**
- * Describes an entity which packages all {@link ProductItem}s of the same type
+ * Describes an entity which packages all {@link life.qbic.datamodel.dtos.business.ProductItem}s of the same type
  *
- * A product is of a defined by a {@link ProductCategory} and other traits that define how to handle items assigned to
+ * A product is of a defined by a {@link life.qbic.datamodel.dtos.business.ProductCategory} and other traits that define how to handle items assigned to
  * that product.
  *
  * @since: 1.0
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/accounting/Product.groovy:11]: Are all the classes from `1.0` rather than `1.12.0` ?

Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/accounting/Product.groovy:11]: I didn't check when the original now deprecated product class was introduced into data-model-lib, but i can check if the versioning is correct tomorrow 👍

**Pull request merged**